src/query/service/src/pipelines/processors/transforms/hash_join/hash_join_build_state.rs

```diff
@@ -933,6 +933,7 @@ impl HashJoinBuildState {
     ) -> Result<()> {
         if !build_key.data_type().remove_nullable().is_numeric()
             && !build_key.data_type().remove_nullable().is_string()
+            && !build_key.data_type().is_date()
```

Comment:
Maybe we can encapsulate them to a method: `runtime_filter_surported_types`.

Reply:
ok